src/kvstore.c

```diff
@@ -0,0 +1,589 @@
+/*
+ * Index-based KV store implementation
+ * This file implements a KV store comprised of an array of dicts (see dict.c)
+ * The purpose of this KV store is to have easy access to all keys that belong
+ * in the same dict (i.e. are in the same dict-index)
+ *
+ * For example, when Redis is running in cluster mode, we use kvstore to save
+ * all keys that map to the same hash-slot in a separate dict within the kvstore
+ * struct.
+ * This enables us to easily access all keys that map to a specific hash-slot.
+ *
+ * Copyright (c) 2024, Redis Labs, Inc
+ * All rights reserved.
```

Comment:
let's take this opportunity to attribute some of this code to Vitaly / hpatro / AWS who invested a lot of effort in dict.c and didn't that line there.
@madolson please advise.

Reply:
I'm against adding this type of copyright for a specific person/company since it's almost always not accurate, since other people contribute. I think we should just say,
```suggestion
 * Copyright (c) 2024, Redis contributors.
 * All rights reserved.
```